### ISSUE TITLE
Longer Term Predictions

### ISSUE THREAD

**Issue by mitkina** [coxlab/prednet#55]
Do you have any advice as to how to get the network to make longer term predictions (e.g., 2 seconds at 10 Hz) without completely deteriorating by the end?

Thanks!

**Comment by bill-lotter** [comment]
Hmm, is this the default model on KITTI? And doing t+1 prediction for each time step? Have you tried training it up to 20 timesteps or is this just in inference mode?

**Comment by mitkina** [comment]
This is not quite on the default KITTI data, but comparable. Doing t+1 works just fine - it's the actual longer term prediction (e.g., recursively predicting for 2 seconds from an input of 0.5 seconds) that deteriorates pretty quickly. I perform the fine-tune as suggested.

Thanks!

**Comment by bill-lotter** [comment]
Ah okay yeah it's probably just that the data gets non-deterministic as time goes on. Would probably need some type of GAN loss for longer term prediction.

**Comment by mitkina** [comment]
Got it. Do you have any context for how far you could push the prediction with the original KITTI data?

**Comment by bill-lotter** [comment]
I would say up to ~5 time steps it can still be okay. There are a few examples at the end of the page here:  https://coxlab.github.io/prednet/. Figure 7 in the paper also has some quantitative info.

**Comment by mitkina** [comment]
Thanks! I was wondering if you tried pushing it any further than 0.5 second predictions. But I guess that's just the limit of the model! Thanks again for your help!

**Comment by xiehousen** [comment]
> Ah okay yeah it's probably just that the data gets non-deterministic as time goes on. Would probably need some type of GAN loss for longer term prediction.

What you are talking about is using the discriminator network as a loss function or using the GAN network to get some loss function.

**Comment by bill-lotter** [comment]
> Thanks! I was wondering if you tried pushing it any further than 0.5 second predictions. But I guess that's just the limit of the model! Thanks again for your help!

No problem! Yeah there isn't a time step where it suddenly jumps up in MSE (it increases pretty linearly like Fig 7 suggests), but ~5 timesteps is a good rule of thumb.

**Comment by bill-lotter** [comment]
> > Ah okay yeah it's probably just that the data gets non-deterministic as time goes on. Would probably need some type of GAN loss for longer term prediction.
> 
> What you are talking about is using the discriminator network as a loss function or using the GAN network to get some loss function.
> Thanks for your reply.

Yeah it would be more along the lines of using a discriminator to add to the loss function.
